src/test/java/bio/terra/cloudres/google/common/GoogleResourceClientHelperTest.java

```
@@ -0,0 +1,67 @@
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test for {@link GoogleResourceClientHelper}
+ */
+@Tag("unit")
+public class GoogleResourceClientHelperTest {
+    private static final String CLIENT = "TestClient";
+    private static final List<TagValue> API_COUNT = Arrays.asList(TagValue.create(CLIENT), TagValue.create(CloudApiMethod.GOOGLE_CREATE_PROJECT.name()));
+    private static final List<TagValue> ERROR_COUNT = Arrays.asList(TagValue.create(CLIENT), TagValue.create(CloudApiMethod.GOOGLE_CREATE_PROJECT.name()), null);
+
+    private GoogleResourceClientHelper helper;
+    private GoogleResourceClientOptions options;
+    private Credentials credentials;
+
+    @Mock
+    private Callable<Project> mockCallable = mock(Callable.class);
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        credentials = NoCredentials.getInstance();
+        options = GoogleResourceClientOptions.Builder.newBuilder().setCredential(credentials).setClient(CLIENT).build();
+        helper = new GoogleResourceClientHelper(options);
+    }
+
+    @Test
+    public void testExecuteGoogleCloudCall_success() throws Exception {
+        helper.executeGoogleCloudCall(mockCallable, CloudApiMethod.GOOGLE_CREATE_PROJECT);
+
+        // One cloud api count
+        assertEquals(AggregationData.CountData.create(1), MetricsHelper.viewManager.getView(MetricsHelper.API_VIEW_NAME).getAggregationMap().get(API_COUNT));
+        // no errors
+        assertNull(MetricsHelper.viewManager.getView(MetricsHelper.ERROR_VIEW_NAME).getAggregationMap().get(ERROR_COUNT));
+    }
+
+    @Test
+    public void testExecuteGoogleCloudCall_withException() throws Exception {
+        when(mockCallable.call()).thenThrow(ResourceManagerException.class);
```

can we just pass in a '() -> throw new ResourceManagereException(); ' instead of introducing mocks?

---

I'd like to mock the callable then we can verify the metrics to make sure the error is being recorded? I feel that closer to what future real test gona looks like as CRL getting complicated.
Also I am looking about some fault injection stuff and trying to find a way to improve this.

---

We don't have to mock the callable, we can pass a lambda that throws whatever we want. The callable doesn't need to be backed by a real google resource client library in anyway. there doesn't need to be Projects here at all. 
This test just verifies that if the callable throws an exception, an error metric is incremented.
I think we might be talking past each other.